pkg/ingester/flush.go

```diff
@@ -110,14 +114,23 @@ func (i *Ingester) sweepUsers(immediate bool) {
 		return
 	}
 
+	oldest := model.Time(0)
+
 	for id, state := range i.userStates.cp() {
 		for pair := range state.fpToSeries.iter() {
 			state.fpLocker.Lock(pair.fp)
 			i.sweepSeries(id, pair.fp, pair.series, immediate)
 			i.removeFlushedChunks(state, pair.fp, pair.series)
+			first := pair.series.firstUnflushedChunkTime()
 			state.fpLocker.Unlock(pair.fp)
+
+			if first > 0 && (oldest == 0 || first < oldest) {
```

Comment:
This would be simplified by initializing `oldest` to maxInt, perhaps?

How does your alerting rule look when the metric is sometimes zero?

Reply:
I think something like this can work (12 hours check):

`(cortex_oldest_unflushed_chunk_timestamp_seconds >bool 0) * (time() - cortex_oldest_unflushed_chunk_timestamp_seconds) > 12*60*60`

Reply:
You're right that it can be simplified by using MaxInt64, but I want to return 0 in case of no chunks. I think that's better than 9.223372036854776e+18, which is math.MaxInt64 converted to float64.